Whiskey/Functions/Resolve-WhiskeyVariable.ps1

```diff
@@ -175,8 +189,89 @@ function Resolve-WhiskeyVariable
                 }
             }
 
-            $needleEnd = $haystack.IndexOf(')', $needleStart)
+            $needleEnd = $needleStart + 2
+            $depth = 0
+            while( $needleEnd -lt $haystack.Length )

```

Comment:
This whole `while` section could use some comments describing what each section is doing. As it stands right now, I'd have a hell of a time debugging some code in this loop.